pkg/reports/reports.go

```diff
@@ -1,13 +1,81 @@
 package reports
 
 import (
+	"fmt"
+
 	"github.com/jetstack/preflight/api"
 	"github.com/jetstack/preflight/pkg/packaging"
 	"github.com/jetstack/preflight/pkg/results"
 	"github.com/jetstack/preflight/pkg/rules"
 	"github.com/jetstack/preflight/pkg/version"
 )
 
+// NewClusterSummary builds a summary for a current cluster based on a
+// freshly generated report set
+func NewClusterSummary(reports []api.Report) (api.ClusterSummary, error) {
+	if len(reports) < 1 {
+		return api.ClusterSummary{}, fmt.Errorf("you must supply at least one report")
+	}
+
+	reportSet, err := NewReportSet(reports)
+	if err != nil {
+		return api.ClusterSummary{}, fmt.Errorf("error constructing report set: %v", err)
+	}
+
+	return api.ClusterSummary{
+		Cluster:         reportSet.Cluster,
+		LatestReportSet: &reportSet,
+	}, nil
+}
+
+// NewReportSet generates a summarized report set from the supplied reports
+func NewReportSet(reports []api.Report) (api.ReportSet, error) {
```

Comment:
A `ReportSet` is intended to contain summaries of reports with **the same timestamp**.

```
// ReportSet groups one or more reports of different packages with the same timestamp for the same cluster.
```

We will use this with the latest set of reports.

Who is going to be responsible for extracting the latest ones? If it is not this function, I'd document that this assumes the reports have the same timestamp. I'd recommend to check them, just to make sure we are not misusing it.

Another option is to make `NewReportSet` extract from the slice of reports the latest set of reports.

Reply:
I've had a go at this in 7dbad8df549194541a0c70afc1c9d6b8d85856e6 - will try and dry up the test data for all these later.